### PR TITLE
Diagnostics group for perturbations from reference state

### LINKED ISSUE #1225: Diagnostic output for field perturbations
# Description

A standard benchmark for evaluating the output of atmospheric dynamical cores are field perturbations. These are fields that describe how far the flow has evolved from the background profile of the simulation; that is, `q - q_background`, where `q` is a prognostic (or diagnostic in the case of potential temperature) field and `q_background` is the corresponding background field.

Based on a discussion with @akshaysridhar, @kpamnany, and @charleskawczynski, this should be relatively straightforward to add.

# Additional context

Having this capability will help us better compare our output with other dynamical cores and analytic results when possible.

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

### PR DESCRIPTION
**Pull request by kpamnany**
# Description

Currently requires that an [`InterpolationConfiguration`](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Driver/diagnostics_configs.jl#L12-L16) be provided.

Add it to your experiment's diagnostics configuration the usual way (e.g. [BOMEX](https://github.com/CliMA/ClimateMachine.jl/blob/master/experiments/AtmosLES/bomex.jl#L464-L479), but with an `InterpolationConfiguration` specified, as in [Held Suarez](https://github.com/CliMA/ClimateMachine.jl/blob/master/experiments/AtmosGCM/heldsuarez.jl#L185-L201)).

Closes #1225.

I have

- [X] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [X] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [X] There are no open pull requests for this already
- [X] CLIMA developers with relevant expertise have been assigned to review this submission
- [X] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by kpamnany** [comment]
This needs a test!

**Comment by kpamnany** [comment]
This is (a little belatedly) ready for merge. @thomasgibson and/or @bischtob: can you check this branch out and take a look at the perturbations produced for Held Suarez?

**Comment by trontrytel** [comment]
@kpamnany - do you know what causes the error in the documentation build in this PR? I think I have a similar problem, but I don't know what that it.

**Comment by kpamnany** [comment]
@trontrytel: I've been trying to figure it out but have no clue yet. Running `docs/make.jl` produces an error running the Literate notebook for Held Suarez, but I'm able to run it interactively with no problem. 😕

File: src/Diagnostics/atmos_refstate_perturbations.jl [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    vars.ref_state.temp = thermo.temp - aux.ref_state.T
+    vars.ref_state.et =
+        (state.ρe / state.ρ) - (aux.ref_state.ρe / aux.ref_state.ρ)
+    # FIXME properly
+    if atmos.moisture isa EquilMoist
+        vars.ref_state.qt =
+            (thermo.moisture.ρq_tot / state.ρ) -
+            (aux.ref_state.ρq_tot / aux.ref_state.ρ)
+    else
+        vars.ref_state.qt = Inf
```
Comment (thomasgibson) [inline src/Diagnostics/atmos_refstate_perturbations.jl:77]: Maybe I'm just misunderstanding here, but if `qt` is the total moisture, shouldn't this zero for a dry model? Setting it to `Inf` here seems unsafe.

Comment (kpamnany) [reply src/Diagnostics/atmos_refstate_perturbations.jl:77]: The standard `missing` value in NetCDF is `NaN` not `Inf` 
(per [this](https://www.unidata.ucar.edu/software/netcdf/workshops/2011/bestpractices/MissingData.html)), I'll fix this thanks.

**Pull request merged**